### PR TITLE
Added SwingingBlade

### PR DESCRIPTION
**Pull request by nihal111**
Adds the swinging blade with a test structural template.

### Steps to test
1. Console: `give toolbox`
2. Select toolbox and rightclick
3. Select "Structure Spawner" -> "AdventureAssets:bladeRoom" -> Give Item
4. Select the Structure Spawner from inventory and right click nearby to spawn it.

Enter the room and witness the gigantic swingingBlade.

### Outstanding
- [x] Make motion of the swinging blade harmonic. (sin or cos).
- [x] Add a new model and attach it to a texture.
- [x] Add collider for the swinging blade and logic for dealing damage to player. (Works in single player)
- [x] Add separate entities for different client side and server side handling.
- [x] Add a separate SpawnSwingingBladeComponent
- [x] Add a trapConfigurationTool that allows to alter trap properties in-game.
- [x] Make the trapConfigurationTool write to the Structural Template prefab.
- [ ] Make everything multiplayer compatible.

### CODE REVIEW

File: src/main/java/org/terasology/adventureassets/traps/swingingblade/SpawnSwingingBladeServerSystem.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            EntityBuilder entityBuilder = entityManager.newBuilder(swingingBlade.prefab);
+            LocationComponent locationComponent = entityBuilder.getComponent(LocationComponent.class);
+            locationComponent.setWorldPosition(position.toVector3f());
+            locationComponent.setWorldRotation(rotation);
+            SwingingBladeComponent swingingBladeComponent = entityBuilder.getComponent(SwingingBladeComponent.class);
+            swingingBladeComponent.timePeriod = swingingBlade.timePeriod;
+            swingingBladeComponent.amplitude = swingingBlade.amplitude;
+            swingingBladeComponent.offset = swingingBlade.offset;
+
+            entityBuilder.build();
```
Comment (flo) [inline src/main/java/org/terasology/adventureassets/traps/swingingblade/SpawnSwingingBladeServerSystem.java:94]: In neither prefab nor during the construction i see any persistence flag setting

File: src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeClientSystem.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    private static final Logger logger = LoggerFactory.getLogger(SwingingBladeClientSystem.class);
+
+    @In
+    private EntityManager entityManager;
+
+    @ReceiveEvent(components = {SwingingBladeComponent.class, LocationComponent.class})
+    public void onSwingingBladeCreated(OnActivatedComponent event, EntityRef entity,
+                                       SwingingBladeComponent swingingBladeComponent) {
+        Prefab swingingBladePrefab = CoreRegistry.get(AssetManager.class).getAsset("AdventureAssets:swingingBladeMesh", Prefab.class).get();
```
Comment (flo) [inline src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeClientSystem.java]: Asset manager schould be obtained via dependency injection

File: src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeClientSystem.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        Location.attachChild(entity, swingingBlade, new Vector3f(Vector3f.zero()), new Quat4f(Quat4f.IDENTITY));
+    }
+
+    @Override
+    public void update(float delta) {
+        for (EntityRef blade : entityManager.getEntitiesWith(SwingingBladeComponent.class)) {
+            LocationComponent locationComponent = blade.getComponent(LocationComponent.class);
+            SwingingBladeComponent swingingBladeComponent = blade.getComponent(SwingingBladeComponent.class);
+            if (locationComponent != null && swingingBladeComponent.isSwinging) {
+                float t = CoreRegistry.get(Time.class).getGameTime();
```
Comment (flo) [inline src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeClientSystem.java]: Time should b obtained via dependenxy inkection like anything else ibtained currently from CoreRegistry

File: src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeComponent.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+package org.terasology.adventureassets.traps.swingingblade;
+
+import org.terasology.entitySystem.Component;
+
+/**
+ * This component holds the data for a Swinging Blade.
+ */
+
+public class SwingingBladeComponent implements Component {
+    /* Time taken by the swinging blade to complete one two and fro motion (in seconds)*/
```
Comment (flo) [inline src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeComponent.java]: A star is missing to make it a javadoc

File: src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeServerSystem.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Override
+    public void update(float delta) {
+        for (EntityRef blade : entityManager.getEntitiesWith(SwingingBladeComponent.class)) {
+            LocationComponent locationComponent = blade.getComponent(LocationComponent.class);
+            SwingingBladeComponent swingingBladeComponent = blade.getComponent(SwingingBladeComponent.class);
+            if (locationComponent != null && swingingBladeComponent.isSwinging) {
+                float t = CoreRegistry.get(Time.class).getGameTime();
+                float T = swingingBladeComponent.timePeriod;
```
Comment (flo) [inline src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeServerSystem.java]: Identifiers of variables should be lower case

**Review by flo — CHANGES_REQUESTED** [review]

File: src/main/java/org/terasology/adventureassets/traps/swingingblade/SpawnSwingingBladeServerSystem.java (hunk 1 above)
Comment (nihal111) [reply src/main/java/org/terasology/adventureassets/traps/swingingblade/SpawnSwingingBladeServerSystem.java:94]: Fixed.
The root entity or the "SwingingBladePlaceholder" entity has the persistent flag in prefab.
The children entities for rod, blade and mesh are `setPersistent(false)` in the SwingingBladeServerSystem and SwingingBladeClientSystem.

File: src/main/java/org/terasology/adventureassets/traps/TrapPlaceholderSystem.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    private AssetManager assetManager;
+    @In
+    private BlockManager blockManager;
+    @In
+    private BlockEntityRegistry blockEntityRegistry;
+    @In
+    private WorldProvider worldProvider;
+
+    @ReceiveEvent
+    public void onBuildTemplateStringWithBlockRegions(BuildStructureTemplateStringEvent event, EntityRef template,
```
Comment (flo) [inline src/main/java/org/terasology/adventureassets/traps/TrapPlaceholderSystem.java]: Method name is wrong

**Review by flo — CHANGES_REQUESTED** [review]

File: src/main/java/org/terasology/adventureassets/traps/TrapPlaceholderSystem.java (hunk 6 above)
Comment (nihal111) [reply src/main/java/org/terasology/adventureassets/traps/TrapPlaceholderSystem.java]: Fixed.

**Pull request merged**